blogs/docker-ci-tricks-2/blog.md

```diff
@@ -71,11 +71,25 @@ There is no definition of a `DIRECTORY_HASH` in here, but there is an execution
 
 ```shell script
 # ... skipped ...
+image_name=$1
 
 DIRECTORY_HASH=$(git rev-parse HEAD:ci/$image_name)
 export DIRECTORY_HASH
-
-# ... skipped ...
+cd ci/$image_name/
+
+# If image is not found by pull, build the image and push it to the Docker Hub.
+docker-compose pull $image_name
+
+image_tag=$(docker-compose config | yq eval ".services.$image_name.image" -)
+docker image inspect "$image_tag" &>/dev/null || {
```

Comment:
As suggested in the issue, I'd simplify the entire thing to:

```suggestion
docker-compose pull $image_name || {
```

This is a bit dishonest, but if someone is interested they'd check the actual sources anyway ;)